context_test.go

```diff
@@ -263,6 +264,49 @@ func TestLoadProjectNoSrcDir(t *testing.T) {
 	}
 }
 
+func TestLoadProjectCfgFileCase(t *testing.T) {
+	if runtime.GOOS != "windows" && runtime.GOOS != "darwin" {
+		t.Skip("skip this test on non-Windows, non-macOS")
+	}
```

Comment:
Can we change this to use `fs.IsCaseSensitiveFilesystem` instead? Since that's what we are testing.

Reply:
I am slightly unsure about this. The function `checkCfgFilenames` uses `fs.IsCaseSensitiveFilesystem` as well.

Reply:
I would say that's fine since `checkCfgFilenames` really depends on `fs.IsCaseSensitiveFilesystem` being correct to function properly. (i.e. a bug in `fs.IsCaseSensitiveFilesystem` will most likely result in `checkCfgFilenames` not behaving properly.

Reply:
But a bug in `fs.IsCaseSensitiveFilesystem` could prevent the test from being run.

Reply:
@ibrasho @sdboyer If you are sure about this, I will do the requested changes. Could you please confirm?

Reply:
I would vote to change this to `fs.IsCaseSensitiveFilesystem`, but I'll leave the final call to @sdboyer . 😄

Reply:
i think, for this _specific_ case (and the other test doing this check in this PR), we should probably prefer to have the test results for `fs.IsCaseSensitiveFilesystem()` and this test vary independently, so it's preferable to use the OS heuristic.

however, we need a comment explaining that this is the specific justification, as this is (or should be) the only place that we prefer the heuristic over doing the actual work of validating fs case sensitivity via `fs.IsCaseSensitiveFilesystem()`.

Reply:
Hey @sdboyer, Could you please help me phrase the comment? Is this good enough - `This is the only place that we prefer the heuristic over doing the actual work of validating fs case sensitivity via fs.IsCaseSensitiveFilesystem()` ? While I get the gist, I am wondering if it can be phrased in simpler words.

Reply:
i struggle with such phrasing things, as well :)

that seems good enough, but it also needs a bit that explains _why_ we're avoiding it here - the reasons you gave earlier, about ensuring that our testing outcomes aren't correlated.